src/app/Http/Controllers/API/AuthController.php

```diff
@@ -106,4 +106,57 @@ public function user(Request $request)
     {
         return new UserResource($request->user());
     }
+  
+  /**
+     * Send an email to the user with the reset password link
+     *
+     * @bodyParam email email required email of the user who makes the request
+     *
+     * @param Request $request
+     */
+    public function sendResetLinkEmail(Request $request)
+    {
+        $request->validate([
+            'email' => 'required|email|exists:users,email'
+        ]);
+
+        // generate and save the token
+        $token = Str::random(60);
+        DB::table('password_resets')->insert([
+            'email' => $request->email,
+            'token' => $token,
+            'created_at' => Carbon::now()
+        ]);
+
+        Mail::to($request->email)->send(new ResetPasswordMail($token));
+    }
+
+    /**
+     * Change the password
+     *
+     * @bodyParam token string required token to validate the request
+     * @bodyParam email email required email of the user who wants to change is password
+     * @bodyParam password string required the new password for the user
+     *
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function reset(Request $request)
+    {
+        $request->validate([
+            'token' => 'required',
+            'password' => 'required|string|min:6',
+            'email' => 'required|email'
+        ]);
+
+        $passwordRowRequest = DB::table('password_resets')->where(['email' => $request->email, 'token' => $request->token]);
```

Comment:
Why require the email when the token (supposedly made out of a CSPRNG) already uniquely identifies the requester and thus the related email ?

I see no scenario where this additional field could yield additional protection against some attack.

Reply:
the tokens is related to a user so if you verify the token with the email you avoid token forgery because you need to have an email with a matching token and not only a token

Reply:
the tokens is related to a user so if you verify the token with the email you avoid token forgery because you need to have an email with a matching token and not only a token

Reply:
I would find it very difficult to forge a token that matches a reset entry, but it doesn't hurt to have additional checks.